[NHT020305] iP

NHT020305 opened the pull request:
# Peter
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Peter frees your mind of having to remember things you need to do. It's,

+ text-based
+ easy to learn
+ ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

``` 
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Huytung020305 committed:
update

Huytung020305 committed:
update

Huytung020305 committed:
update

Huytung020305 committed:
update

Huytung020305 committed:
update

Huytung020305 committed:
Level-5

Huytung020305 committed:
update

Huytung020305 committed:
update

Huytung020305 committed:
level-7

Huytung020305 committed:
merge branch-Level-7

Huytung020305 committed:
merge branch-Level-8

Huytung020305 committed:
A-MoreOOP

Huytung020305 committed:
A-Packages

Huytung020305 committed:
update

Huytung020305 committed:
Merge branch 'add-gradle-support'

Huytung020305 committed:
update

Huytung020305 committed:
A-Gradle

Huytung020305 committed:
A-JUnit

Huytung020305 committed:
A-Jar

Huytung020305 committed:
A-JavaDoc

Huytung020305 committed:
Level-9

Huytung020305 committed:
Level-9

Huytung020305 committed:
Merge branch 'branch-Level-9'

Huytung020305 committed:
Update project

xenyeo commented on src/main/java/Peter/Peter.java:
```diff
@@ -0,0 +1,69 @@
+                String fullCommand = ui.readCommand();
+                ui.showLine();
+                Command command = new CommandParser()
+                                        .makeSenseUserCommand(fullCommand);
```
```suggestion
        .makeSenseUserCommand(fullCommand);
```

xenyeo replied:
Indentation for wrapped lines should be 8 spaces.

xenyeo replied:
Indentation for wrapped lines should be 8 spaces.

xenyeo commented on src/main/java/Peter/Peter.java:
```diff
@@ -0,0 +1,69 @@
+    public static void main(String[] args) {
+        new Peter("./data/Peter.txt").run();
+    }
+}
```
LGTM.

xenyeo commented on src/main/java/Peter/command/Command.java:
```diff
@@ -0,0 +1,34 @@
+     */
+    public abstract void execute(Ui ui, TaskManager taskManager,
+                                 TaskStorage taskStorage);
+
```
This method signature can be written in a single line.

xenyeo commented on src/main/java/Peter/command/Command.java:
```diff
@@ -0,0 +1,34 @@
+     * @return true if the program should terminate, false otherwise.
+     */
+    public abstract boolean isTerminal();
+}
```
LGTM.

xenyeo commented on src/main/java/Peter/command/CommandParser.java:
```diff
@@ -0,0 +1,54 @@
+        throw new MeaninglessCommandException(
+                "OOPS!!! I'm sorry, but I don't know what that means :-(");
+    }
+}
```
LGTM.

xenyeo commented on src/main/java/Peter/command/commands/AddCommand.java:
```diff
@@ -0,0 +1,28 @@
+import Peter.task.TaskManager;
+import Peter.ui.Ui;
+
+public class AddCommand extends Command {
```
Missing header comment for public class, if not LGTM.

xenyeo commented on src/main/java/Peter/command/commands/ByeCommand.java:
```diff
@@ -0,0 +1,17 @@
+import Peter.task.TaskManager;
+import Peter.ui.Ui;
+
+public class ByeCommand extends Command {
```
Missing header comment for public class, if not LGTM.

xenyeo commented on src/main/java/Peter/command/commands/CountCommand.java:
```diff
@@ -0,0 +1,24 @@
+import Peter.task.TaskManager;
+import Peter.ui.Ui;
+
+public class CountCommand extends Command {
```
Missing header comment for public class, if not LGTM.

xenyeo commented on src/main/java/Peter/command/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,28 @@
+import Peter.task.TaskManager;
+import Peter.ui.Ui;
+
+public class DeleteCommand extends Command {
```
Missing header comment for public class, if not LGTM.

xenyeo commented on src/main/java/Peter/command/commands/FindCommand.java:
```diff
@@ -0,0 +1,25 @@
+import Peter.task.TaskManager;
+import Peter.ui.Ui;
+
+public class FindCommand extends Command{
```
Missing header comment for public class, if not LGTM.

xenyeo commented on src/main/java/Peter/command/commands/InstructionCommand.java:
```diff
@@ -0,0 +1,17 @@
+import Peter.task.TaskManager;
+import Peter.ui.Ui;
+
+public class InstructionCommand extends Command {
```
Missing header comment for public class, if not LGTM.

xenyeo commented on src/main/java/Peter/command/commands/ListCommand.java:
```diff
@@ -0,0 +1,23 @@
+import Peter.task.TaskManager;
+import Peter.ui.Ui;
+
+public class ListCommand extends Command {
```
Missing header comment for public class, if not LGTM.

xenyeo commented on src/main/java/Peter/command/commands/MarkCommand.java:
```diff
@@ -0,0 +1,26 @@
+import Peter.task.TaskManager;
+import Peter.ui.Ui;
+
+public class MarkCommand extends Command {
```
Missing header comment for public class, if not LGTM.

xenyeo commented on src/main/java/Peter/command/commands/ResetCommand.java:
```diff
@@ -0,0 +1,20 @@
+import Peter.task.TaskManager;
+import Peter.ui.Ui;
+
+public class ResetCommand extends Command {
```
Missing header comment for public class, if not LGTM.

Huytung020305 committed:
Update project

xenyeo commented on src/main/java/Peter/command/commands/UnMarkCommand.java:
```diff
@@ -0,0 +1,26 @@
+import Peter.task.TaskManager;
+import Peter.ui.Ui;
+
+public class UnMarkCommand extends Command {
```
Missing header comment for public class, if not LGTM.
`UnMarkCommand` can be renamed to `UnmarkCommand` for clarity since unmark is a word on its own.

xenyeo commented on src/main/java/Peter/storage/TaskGenerator.java:
```diff
@@ -0,0 +1,82 @@
+     * Constructs a TaskGenerator.
+     */
+    public TaskGenerator() {
+    }
```
Shouldn't this be omitted if its empty?

xenyeo reviewed (commented):
Overall code looks good, LGTM in general.

Huytung020305 committed:
Update project

Huytung020305 committed:
Update project

Huytung020305 committed:
Update project

kennethsoh commented on src/main/java/Peter/command/CommandParser.java:
```diff
@@ -0,0 +1,54 @@
+package Peter.command;
+
+import Peter.command.commands.*;
```
I think imported classes should be listed explicitly

kennethsoh commented on src/main/java/Peter/command/CommandParser.java:
```diff
@@ -0,0 +1,54 @@
+    public Command makeSenseUserCommand(String command) throws MeaninglessCommandException,
+            EmptyTaskException, InvalidTaskFormatException, InvalidDateTimeFormatException {
+        if (command.equals("bye")) {
+            return new ByeCommand();
```
For all the methods referenced in this file, I think that camelCase should be used.

kennethsoh commented on src/main/java/Peter/task/Task.java:
```diff
@@ -0,0 +1,81 @@
+    public String toString() {
+        return String.format("[%s] %s", this.getStatusIcon(), this.description);
+    }
+}
```
LGTM 👍🏻

Huytung020305 committed:
Update project

Huytung020305 committed:
Update project

Huytung020305 committed:
Update project

Huytung020305 committed:
Update project

Huytung020305 committed:
Update project

Huytung020305 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	src/main/java/peter/Peter.java

Huytung020305 committed:
Update project

Huytung020305 committed:
Update project

Huytung020305 committed:
Update project

Huytung020305 committed:
Update project

Huytung020305 committed:
Update project

elizazaa commented on src/main/java/peter/datetime/LocalDateTimeParser.java:
```diff
@@ -0,0 +1,76 @@
+     * @throws ArrayIndexOutOfBoundsException If the input string is not in the expected format.
+     * @throws NumberFormatException If date or time components cannot be parsed as integers.
+     */
+    public LocalDateTime convertToTime() throws InvalidDateTimeFormatException {
```
Too Long Method

Huytung020305 committed:
A-Assertions

Huytung020305 committed:
A-CodeQuality

Huytung020305 committed:
Update project

Huytung020305 committed:
Add the assumption in add method that the task is not null

Huytung020305 committed:
Update project

Huytung020305 committed:
Add assert statement in initialize() method

Huytung020305 committed:
Merge master into branch-A-Assertions

NHT020305 committed:
Merge pull request #2 from NHT020305/branch-A-Assertions

A-Assertions

Huytung020305 committed:
Merge branch 'master' into branch-A-CodeQuality

NHT020305 committed:
Merge pull request #3 from NHT020305/branch-A-CodeQuality

Merge A-CodeQuality

Huytung020305 committed:
Add C-Update extension

Huytung020305 committed:
C-Update extension: update different kinds of task information

Users do not want to rewrite the whole task information

Allowing users to update a piece of information of a task is more convenient

Let's update the algorithm in UpdateCommand class, Task class, Todo class, Deadline class and Event class so that we can update different kinds of information of any task.

Huytung020305 committed:
Improve code quality

There are too many magic literals

Assign them to static variables will make the code easier to understand

Divide them into different classes that only handle magic literal: CommandType, TaskKeyword, ErrorMessage, DateTime, and ReplyMessage

Huytung020305 committed:
Improve code quality

There are too many magic literals

Assign them to static variables will make the code easier to understand

Divide them into different classes that only handle magic literal: CommandType, TaskKeyword, ErrorMessage, DateTime, and ReplyMessage

NHT020305 committed:
Update README.md

NHT020305 committed:
Update README.md

NHT020305 committed:
Update README.md

Huytung020305 committed:
Add Ui.png

NHT020305 committed:
Update README.md

Huytung020305 committed:
Update README.md

NHT020305 committed:
Update README.md

NHT020305 committed:
Update README.md

NHT020305 committed:
Update README.md

NHT020305 committed:
Update README.md

NHT020305 committed:
Update README.md

NHT020305 committed:
Update README.md

NHT020305 committed:
Update README.md

NHT020305 committed:
Update README.md

NHT020305 committed:
Update README.md

Huytung020305 committed:
Update new Peter.png

Huytung020305 committed:
Update project docs

Huytung020305 committed:
Update project README.md

Huytung020305 committed:
Update project README.md

Huytung020305 committed:
Update project README.md

Huytung020305 committed:
Update methods to meet SLAP principle

Huytung020305 committed:
Update README.md

Huytung020305 committed:
Update README.md

Huytung020305 committed:
Update README.md

Huytung020305 committed:
Update README.md

Huytung020305 committed:
Update README.md

Huytung020305 committed:
Update README.md

Huytung020305 committed:
Update README.md
